### DISCUSSION TITLE
is there a limit on graphql response size

### DISCUSSION THREAD

**Discussion by 1zg12** [Netflix/dgs-framework#1070] [General]
I have an implementation leverage on dgs, which could server a response with varied data size.

At times, when the data size above 2GB, it would fail the process. 
![image](https://user-images.githubusercontent.com/52658745/170165334-ba6441e0-f6b1-449f-a7a5-9f7431122ce8.png)

I didn't see any specs saying there is a hard limit of 2GB in dgs. However, I guess with current implementation it does seem like that's the limit. 

Should this be included in the spec, or maybe the implementation should be updated ?

**Comment by srinivasankavitha** [comment]
That's interesting. We haven't run into this issue so far, perhaps we've not had such a huge response to handle. That said, we don't impose any specific limits in the framework. Sound like it is `jackson` that is having the problem when using the object mapper?

**Reply by 1zg12** [reply]
I thought the same in the beginning as well, then `jackson` basically saying `RestController` shouldn't call `writeValueAsString` here.
https://github.com/Netflix/dgs-framework/blob/866f1d4e76b1b85d95661576cb7c027c7d9b1004/graphql-dgs-spring-webmvc/src/main/kotlin/com/netflix/graphql/dgs/mvc/DgsRestController.kt#L211

https://github.com/FasterXML/jackson-core/issues/760

🤦

**Reply by srinivasankavitha** [reply]
OhI see. Thanks for providing the context. We can look into a fix. However, I'm not sure if we do want to handle larger than 2GB responses 🤔

**Reply by 1zg12** [reply]
Personally, I think 2GB is no longer as large as it sounded like before. It's a 20 seconds transport time even for a 100MBps bandwidth.
